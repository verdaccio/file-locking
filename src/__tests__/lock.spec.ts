import path from 'path';
import fs from 'fs';
import { lockFile, unlockFile, readFile } from '../index';

const getFilePath = (filename: string) => {
  return path.resolve(__dirname, `assets/${filename}`);
};

const removeTempFile = (filename: string) => {
  const filepath = getFilePath(filename);
  fs.unlink(filepath, (error) => {
    if (error) throw error;
  });
};

describe('testing locking', () => {
  test('file should be found to be locked', (done) => {
    lockFile(getFilePath('package.json'), (error: any) => {
      expect(error).toBeNull();
      removeTempFile('package.json.lock');
      done();
    });
  });

  test('file should fail to be found to be locked', (done) => {
    lockFile(getFilePath('package.fail.json'), (error: any) => {
      expect(error.message).toMatch(
        /ENOENT: no such file or directory, stat '(.*)package.fail.json'/
      );
      done();
    });
  });

  test('file should to be found to be unLock', (done) => {
    unlockFile(getFilePath('package.json.lock'), (error: any) => {
      expect(error).toBeNull();
      done();
    });
  });

  test('read file with no options should to be found to be read it as string', (done) => {
    readFile(getFilePath('package.json'), (error: any, data: any) => {
      expect(error).toBeNull();
      expect(data).toMatchSnapshot();
      done();
    });
  });

  test('read file with no options should to be found to be read it as object', (done) => {
    const options = {
      parse: true,
    };
    readFile(getFilePath('package.json'), options, (error: any, data: any) => {
      expect(error).toBeNull();
      expect(data).toMatchSnapshot();
      done();
    });
  });

  test('read file with options (parse) should to be not found to be read it', (done) => {
    const options = {
      parse: true,
    };
    readFile(getFilePath('package.fail.json'), options, (error: any) => {
      expect(error.message).toMatch(
        /ENOENT: no such file or directory, open '(.*)package.fail.json'/
      );
      done();
    });
  });

  test('read file with options should to be found to be read it and fails to be parsed', (done) => {
    const options = {
      parse: true,
    };
    readFile(getFilePath('wrong.package.json'), options, (error: any) => {
      expect(error.message).toEqual(
        'Unexpected token } in JSON at position 44'
      );
      done();
    });
  });

  test('read file with  options (parse, lock) should to be found to be read it as object', (done) => {
    const options = {
      parse: true,
      lock: true,
    };
    readFile(getFilePath('package2.json'), options, (error: any, data: any) => {
      expect(error).toBeNull();
      expect(data).toMatchSnapshot();
      removeTempFile('package2.json.lock');
      done();
    });
  });

  test('read file with options (parse, lock) should to be found to be read it and fails to be parsed', (done) => {
    const options = {
      parse: true,
      lock: true,
    };
    readFile(getFilePath('wrong.package.json'), options, (error: any) => {
      expect(error.message).toEqual(
        'Unexpected token } in JSON at position 44'
      );
      removeTempFile('wrong.package.json.lock');
      done();
    });
  });
});
