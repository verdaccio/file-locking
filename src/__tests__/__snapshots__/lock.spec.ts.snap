// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing locking read file with  options (parse, lock) should to be found to be read it as object 1`] = `
Object {
  "name": "assets",
  "version": "0.0.1",
}
`;

exports[`testing locking read file with no options should to be found to be read it as object 1`] = `
Object {
  "name": "assets",
  "version": "0.0.1",
}
`;

exports[`testing locking read file with no options should to be found to be read it as string 1`] = `
"{
  \\"name\\": \\"assets\\",
  \\"version\\": \\"0.0.1\\"
}
"
`;
